crates/lsp/src/lsp.rs

```diff
@@ -332,11 +332,33 @@ impl LanguageServer {
             };
 
             let header = std::str::from_utf8(&buffer)?;
-            let message_len: usize = header
+            let mut segments = header.lines();
+
+            let message_len: usize = segments
+                .next()
+                .ok_or_else(|| anyhow!("unable to find the first line of the LSP message header"))? // Impossible, but just in case.
                 .strip_prefix(CONTENT_LEN_HEADER)
                 .ok_or_else(|| anyhow!("invalid LSP message header {header:?}"))?
-                .trim_end()
-                .parse()?;
+                .parse()
+                .context("failed to parse Content-Length of LSP message header")?;
```

Comment:
Let's use `with_context` and actually print the erroneous header in the message.

Reply:
This one still on, though.
Sorry for nitpicking, but here you change all LSP-related code, so let's do one more round and actually use `.with_context(|| format!(...` and mention what header exactly did we fail to parse.